Regression: Building on macOS fails with a type mismatch

Building fails with the following mismatched type error: 
```
   Compiling kernel v0.1.0 (/Users/user/Downloads/Test Folder/obliteration/src/kernel)
libunwind: malformed __unwind_info at 0x186A17CA8 bad second level page
libunwind: malformed __unwind_info at 0x186A17CA8 bad second level page
error[E0308]: mismatched types
   --> kernel/src/main.rs:230:20
    |
60  | fn start() -> Result<(), KernelError> {
    |               ----------------------- expected `Result<(), KernelError>` because of return type
...
230 |             return ExitCode::FAILURE;
    |                    ^^^^^^^^^^^^^^^^^ expected `Result<(), ...>`, found `ExitCode`
    |
    = note: expected enum `Result<(), KernelError>`
             found struct `ExitCode`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `kernel` (bin "obkrnl") due to previous error
gmake[2]: *** [src/CMakeFiles/core.dir/build.make:95: src/core-prefix/src/core-stamp/core-build] Error 101
gmake[1]: *** [CMakeFiles/Makefile2:102: src/CMakeFiles/core.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
Build failed
```


Full output: 
[Oblit Output.txt](https://github.com/obhq/obliteration/files/14002135/Oblit.Output.txt)

Hardware: MacBook Pro
CPU/GPU: M1 Pro
OS: Sonoma 14.3 RC

--- a/src/kernel/src/main.rs
+++ b/src/kernel/src/main.rs
@@ -31,7 +31,7 @@ use std::sync::Arc;
 use std::time::SystemTime;
 use sysinfo::{MemoryRefreshKind, System};
 use thiserror::Error;
-use tty::TtyError;
+use tty::TtyInitError;
 
 mod arch;
 mod arnd;
@@ -227,7 +227,7 @@ fn start() -> Result<(), KernelError> {
         #[cfg(not(target_arch = "x86_64"))]
         ExecutionEngine::Native => {
             error!("Native execution engine cannot be used on your machine.");
-            return ExitCode::FAILURE;
+            Err(KernelError::NativeExecutionEngineNotSupported)
         }
         ExecutionEngine::Llvm => run(
             args.debug_dump,
@@ -508,8 +508,12 @@ enum KernelError {
     #[error("memory manager initialization failed")]
     MemoryManagerInitFailed(#[from] MemoryManagerError),
 
+    #[cfg(not(target_arch = "x86_64"))]
+    #[error("the native execution engine is only supported on x86_64")]
+    NativeExecutionEngineNotSupported,
+
     #[error("tty initialization failed")]
-    TtyInitFailed(#[from] TtyError),
+    TtyInitFailed(#[from] TtyInitError),
 
     #[error("virtual process initialization failed")]
     VProcInitFailed(#[from] VProcInitError),

--- a/src/kernel/src/tty/mod.rs
+++ b/src/kernel/src/tty/mod.rs
@@ -16,7 +16,7 @@ pub struct TtyManager {
 impl TtyManager {
     const TIOCSCTTY: IoCmd = IoCmd::io(b't', 97);
 
-    pub fn new(fs: &Arc<Fs>) -> Result<Arc<Self>, TtyError> {
+    pub fn new(fs: &Arc<Fs>) -> Result<Arc<Self>, TtyInitError> {
         // Create /dev/console.
         let console = Arc::new(CdevSw::new(
             DriverFlags::from_bits_retain(0x80000004),
@@ -35,7 +35,7 @@ impl TtyManager {
             MakeDev::MAKEDEV_ETERNAL,
         ) {
             Ok(v) => v,
-            Err(e) => return Err(TtyError::CreateConsoleFailed(e)),
+            Err(e) => return Err(TtyInitError::CreateConsoleFailed(e)),
         };
 
         Ok(Arc::new(Self { console }))
@@ -52,9 +52,9 @@ impl TtyManager {
     }
 }
 
-/// Represents an error when [`TtyManager`] was failed to initialized.
+/// Represents an error when [`TtyManager`] fails to initialize.
 #[derive(Debug, Error)]
-pub enum TtyError {
+pub enum TtyInitError {
     #[error("cannot create console device")]
     CreateConsoleFailed(#[source] MakeDevError),
 }

